tempesta_fw/t/unit/test_hash.c

```diff
@@ -63,27 +63,26 @@ TEST(tfw_hash_str, calcs_diff_hash_for_diff_str)
 TEST(tfw_hash_str, calcs_same_hash_for_diff_chunks_n)
 {
 	unsigned long h1, h2, h3;
-
-	TfwStr s1 = { .len = 17, .ptr = (void *)"Host: example.com" };
-
-	TfwStr s2c1 = {	.len = 14, .ptr = (void *)"Host: example." };
-	TfwStr s2c2 = {	.len = 3, .ptr = (void *)"com" };
+	TfwStr s1 = { .len = 17, .data = "Host: example.com" };
+	TfwStr s2c1 = {	.len = 14, .data = "Host: example." };
+	TfwStr s2c2 = {	.len = 3, .data = "com" };
 	TfwStr s2chunks[] = { s2c1, s2c2 };
 	TfwStr s2 = {
+
 		.len = 14 + 3,
```

Comment:
This is very error-prone. A better solution is to use the actual string lengths: `s2c1.len + s2c2.len`.